spec/services/open_library_service.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe OpenLibraryService, type: :service do
+  describe 'GET #isbn_hash' do
+    let!(:service) { OpenLibraryService.new }
+    let(:hash) { service.isbn_hash('0385472579') }
+    context 'when service connects successfully with external API' do
+      it "its response's title is Zen speaks" do
+        expect(hash[:title]).to eq 'Zen speaks'
+      end
+    end
+
+    context "When service fails to connect to external API's URI" do
+      it 'raises and SocketError exception' do
+        allow(OpenLibraryService).to receive(:get).and_raise(SocketError)
+        response = service.isbn_hash('0385472579')
```

Comment:
This kind of things should be placed outside the it, but inside this context.

The it should only have the expect. You can place those in a before block